crates/matrix-sdk-ui/src/timeline/reactions.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 The Matrix.org Foundation C.I.C.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use ruma::{EventId, OwnedEventId, OwnedTransactionId, TransactionId};
+
+#[derive(Clone, Debug)]
+/// The result of toggling a reaction
+///
+/// Holds the data required to update the state of the reaction in the timeline
```

Comment:
Documentation before attributes, please.

Reply:
[`02c5aa9`](https://github.com/matrix-org/matrix-rust-sdk/pull/2118/commits/02c5aa9358b83acbd88409d69a6ab784cbb97944)

Reply:
Looking at the full diff, the attributes are still / once again before the docs 😅

Reply:
Oops - fixed!